File: opencode/aspnet/rawdog/LooseNotes/Models/Rating.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LooseNotes.Models;

public class Rating
{
    public int Id { get; set; }
    
    [Required]
    [Range(1, 5)]
    public int Value { get; set; }
    
    [StringLength(500)]
    public string? Comment { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    [Required]
    public string UserId { get; set; } = string.Empty;
    
    [ForeignKey("UserId")]
    public ApplicationUser User { get; set; } = null!;
    
    public int NoteId { get; set; }
    
    [ForeignKey("NoteId")]
    public Note Note { get; set; } = null!;
}

File: claude/jsp/securable/LooseNotes/src/main/java/com/loosenotes/model/Rating.java
package com.loosenotes.model;

import java.time.Instant;

/**
 * Represents a user's rating (1-5 stars) on a note.
 *
 * SSEM notes:
 * - Integrity: stars range constraint validated in service layer AND DB CHECK constraint.
 */
public class Rating {

    private long id;
    private long noteId;
    private long userId;
    /** Denormalized for display. */
    private String raterUsername;
    /** 1–5 inclusive; validated at service layer. */
    private int stars;
    private String comment;
    private Instant createdAt;
    private Instant updatedAt;

    public Rating() {}

    public long getId() { return id; }
    public void setId(long id) { this.id = id; }

    public long getNoteId() { return noteId; }
    public void setNoteId(long noteId) { this.noteId = noteId; }

    public long getUserId() { return userId; }
    public void setUserId(long userId) { this.userId = userId; }

    public String getRaterUsername() { return raterUsername; }
    public void setRaterUsername(String raterUsername) { this.raterUsername = raterUsername; }

    public int getStars() { return stars; }
    public void setStars(int stars) { this.stars = stars; }

    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }

    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }

    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }

    @Override
    public String toString() {
        return "Rating{id=" + id + ", noteId=" + noteId + ", userId=" + userId + ", stars=" + stars + '}';
    }
}
